### ISSUE TITLE
setSplitterPosition doesn't change split position inside a `display:block` container

### ISSUE THREAD

**Issue by OlliTietavainenVaadin** [vaadin/flow-components#1002]
### Description
In some conditions, calling `SplitLayout::setSplitterPosition` doesn't do anything. This seems to occur when you have a parent container that has the `display` CSS property set to `block` (which Vaadin Start Views have by default).

### Expected outcome
Calling `setSplitterPosition` should react to the input value.

### Actual outcome
`setSplitterPosition` doesn't do anything.

### Live Demo
Example view: 
```
public class HelloWorldView extends HorizontalLayout {

    public HelloWorldView() {
        getElement().getStyle().set("display", "block");
        SplitLayout sl = new SplitLayout(new Label("First"), new Label("Second"));
        sl.setOrientation(SplitLayout.Orientation.VERTICAL);
        Button hideSecond = new Button("Hide Second", e -> sl.setSplitterPosition(100));
        Button hideFirst = new Button("Hide First", e -> sl.setSplitterPosition(0));
        add(sl, hideSecond, hideFirst);
    }

}
```

### Steps to reproduce
1. Run the app
2. Click either of the buttons
3. Nothing happens
4. Comment out the row that sets the `display: block` style
5. Try again

### Browsers Affected

- [x ] Chrome (only tested browser)

Tested with Vaadin 14.5.5 and 19.0.7, reproduces in both

**Comment by sissbruecker** [comment]
The issue here is not related to the parent being a block container, but comes from the fact that the SplitLayout does not have an explicit height. When using `setSplitterPosition`, it sets a percentage height for the children. However the SplitLayout's height is not explicit, and because of that is calculated based on the contents of the children themselves. This creates a kind of loop where the child wants to be some percentage of the height of the parent, but the parent needs the childrens height to calculate its own height. This SO answer explains it pretty well I think: https://stackoverflow.com/a/1622097/11368281

We discussed changing the logic to calculate the child sizes based on the effective height of the combined childrens contents, but that would lead to even more edge-cases and non-deterministic behaviour.

As such we decided to not support this use-case, and instead add a note to the docs that a vertical SplitLayout needs a well-defined height, either as an absolute value or a percentage. Note that when using a percentage height on the SplitLayout, it's ancestor(s) needs to have an explicitely defined height too.

@OlliTietavainenVaadin Regarding the start.vaadin.com templates, I can offer to take a look at what the problem is there and how we can change them to provide a well-defined height for the SplitLayout.

It is also not 100% clear what the original use-case was, it kind of looks like the intention was to toggle the visibility of one child of the SplitLayout. If that is the case: SplitLayout only works with exactly two children ATM. A workaround could be to write a wrapper component, that contains the SplitLayout and replaces the SplitLayout with one of its children when toggling. Here is a similar setup that we use for a test: https://github.com/vaadin/flow-components/blob/master/vaadin-split-layout-flow-parent/vaadin-split-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/splitlayout/test/LayoutComponent.java

I will take care of updating the docs and close this ticket after that is resolved.
